torch_geometric/loader/link_neighbor_loader.py

```diff
@@ -11,7 +11,42 @@
 
 
 class LinkNeighborSampler(NeighborSampler):
+    def __init__(self, data: Union[Data, HeteroData],
```

Comment:
this could be put in the base class.. but I don't know if negative sampling makes sense for node sampler

Reply:
It looks good how it is. Alternatively, we could add `kwargs` support to `NeighborSampler` that just stores any additionally given arguments:
```python
for key, value in kwargs.items():
    setattr(self, key, value)
```

Reply:
I slightly prefer what I've got so will leave it here. Makes it a bit more explicit what properties can be expected.